src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseTaskDataObject.cs

```diff
@@ -130,7 +142,14 @@ public RestoreOptions RestoreOptions
         /// </summary>
         public string DataFilesFolder
         {
-            get { return this.dataFilesFolder; }
+            get
+            {
+                if (string.IsNullOrEmpty(this.dataFilesFolder))
```

Comment:
this getter uses a slightly different pattern then the one you used in the other getter you added on line 44.  I'd recommend using the same pattern in general for consistency.